plugins/modules/helm.py

```diff
@@ -612,17 +634,38 @@ def main():
         if chart_version is not None:
             helm_cmd += " --version=" + chart_version
 
-        if chart_repo_url is not None:
+        if chart_repo_url:
             helm_cmd += " --repo=" + chart_repo_url
 
         # Fetch chart info to have real version and real name for chart_ref from archive, folder or url
         chart_info = fetch_chart_info(module, helm_cmd, chart_ref)
 
+        if dependency_update:
+            if not chart_info.get('dependencies'):
+                msg_fail = (f"No subchart will be pulled for {chart_ref}. "
+                            "Please make sure to add dependencies block in Chart.yaml or requirements.yaml. "
+                            "For more information please visite https://helm.sh/docs/helm/helm_dependency/")
+                module.fail_json(msg=msg_fail)
+
+            # Can't use '--dependency-update' with 'helm upgrade' that is the
+            # default chart install method, so if chart_repo_url is defined
+            # we can't use the dependency update command. But, in the near future
+            # we can get rid of this method and use only '--dependency-update'
+            # option. Please see https://github.com/helm/helm/pull/8810
+            if not chart_repo_url:
+                run_dep_update(module, helm_cmd_common, chart_ref)
+            else:
+                module.warn("This is a not stable feature with 'chart_repo_url'. Please consider to use dependency update with on-disk charts")
+                if not replace:
+                    msg_fail = ("'--dependency-update' hasn't been supported yet with 'helm upgrade'. "
+                                "Please use 'helm install' instead by adding 'replace' option")
+                    module.fail_json(msg=msg_fail)
```

Comment:
This part is becoming very confusing for me. for 2 reasons:
1. What I found in Ansible doc is that if I make `dependency_update` and `replace` mutually exclusive I can't enable both of them and that is not the case because when the `chart_ref_url` is defined or `chart_ref` is an URL I need to enable `replace` option when `dependency_update` is enabled to use `helm install` instead of `helm update`(not support `--dependency-update` option. I'm sorry, this is my first ansible module creation maybe I need more details about your suggestion.
2. After doing a deep search I realised that we haven't the need to do `dependency update` with remote charts because to package them we need to pull the subchart by running `helm package --dependency-update dep-up` So, if I'm not mistaken, all remote package that depends on other charts have their subchart pulled already within them.